### ISSUE TITLE
Unit tests

### ISSUE THREAD

**Issue by faheel** [ProAlgos/ProAlgos-Cpp#41]
Add unit tests for the following programs:
- [x] N-Queens
- [x] Kadane
- [x] LCS
- [x] KMP
- Searching
	- [x] Common test for linear search and binary search 
	- [ ] Ternary search* 
- Sorting
	- [x] Common test
	- [ ] Radix sort

\* Ternary search currently has a unit test but it needs to be refactored.

**Comment by faheel** [comment]
Read about [Adding unit tests for C++ code](../blob/master/C++/UNIT_TESTS.md) for step-wise directions.

**Comment by faheel** [comment]
Issue #74 is now closed

**Comment by alxmjo** [comment]
All of the **searching** algorithms currently have unit tests, but they don't have a _common_ unit test. However, there is a common test for **sorting**, which may be useful for implementing a common searching unit test.

**Comment by alxmjo** [comment]
@faheel Regarding a common test for searching algorithms, is it your belief that this should include ternary search? The interface and use of ternary search differs from the other two existing search algorithms (linear and binary), which makes me wonder if it would be more straightforward to have a common test for binary and linear (and similar algorithms which may be added later) and then a separate test for ternary search. Not as clean as the sorting implementation I added earlier, but maybe a better fit for the objective at hand.

**Comment by faheel** [comment]
@alxmjo Having a common unit test for linear and binary search and a separate one for ternary search would be best.

**Comment by ghost** [comment]
Hey, I'd love to give this a crack. Let me know if the issue is still open for any left over algorithms.

**Comment by alxmjo** [comment]
@safderareepattamannil If you're still interested, we're still missing a common test for searching algorithms. Have a look at the common test for sorting algorithms for an idea of how you might implement it.

**Comment by beardbytes** [comment]
@alxmjo Hey, I am interested to give it a try !

**Comment by amrii1** [comment]
Hello,
Is still any unit tests that needs to be done

**Comment by alxmjo** [comment]
@amrii1 Yes! See the post at the top of the thread. Currently still need a common test for linear and binary search, and then standalone test files for ternary search and radix sort. Please create a new issue if you have questions or would like to discuss implementation. 🙂

**Comment by abdulwahab-alobaid-2191115290** [comment]
hello the unit testing for the searching algorithms is checked in the readme.md , so is it still open or not ?
i am planning to work on that

**Comment by faheel** [comment]
@abdulwahab-alobaid-2191115290 The unit tests for searching algorithms have already been added. You can pick up something else if you like.

**Comment by lydiathiel** [comment]
I'm planning to submit testing for the radix search within a week or two.

**Comment by Ashu-1309** [comment]
What I have to do?? 
Do i have to write code???

**Comment by srsonia2506** [comment]
@faheel @alxmjo Would love to do unit test for Radix Sort

**Comment by faheel** [comment]
A fix for radix sort has been added in #420. Feel free to review the code and run the tests (and suggest new edge cases if any).
